src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -903,25 +899,11 @@ public abstract partial class WebRequestPSCmdlet : PSCmdlet
 
         private HttpMethod GetHttpMethod(WebRequestMethod method)
         {
-            switch (Method)
+            switch (method)
```

Comment:
I wonder why do we need to remove mappings and use default case with allocation?
I don't see value in the change.

Also the method should be static.

Reply:
The code looked cleaner without them and we already didn't have mappings for Merge and Patch methods, so I didn't see any harm in removing all the others.

Reply:
Although it's not the hot path I always prefer to stick to the simple rule - if we can not do allocations, it's better not to do them. This is especially important for code that will be used by others in scenarios that we don't know.
So I think this change is unwarranted.
As for the missing values, it is better to discuss this in a new issue to see if it is really often used to have them explicitly.

Reply:
Would it be better if I rewrote GetHttpMethod like this and added Patch to the mapping ?

```cs
        private static HttpMethod GetHttpMethod(WebRequestMethod method) => method switch
        {
            WebRequestMethod.Default or
            WebRequestMethod.Get => HttpMethod.Get,
            WebRequestMethod.Head => HttpMethod.Head,
            WebRequestMethod.Post => HttpMethod.Post,
            WebRequestMethod.Put => HttpMethod.Put,
            WebRequestMethod.Delete => HttpMethod.Delete,
            WebRequestMethod.Trace => HttpMethod.Trace,
            WebRequestMethod.Options => HttpMethod.Options,
            WebRequestMethod.Patch => HttpMethod.Patch,
            _ => new HttpMethod(method.ToString().ToUpperInvariant()), //Merge
        };
```

Reply:
Looks good.